File: examples/python/create_intent.py
"""Create a payment intent (stdlib only).

    DASH_AUTH_SECRET=... python examples/python/create_intent.py
"""
import json
import os
import urllib.request

SERVICE = os.environ.get("DASH_SERVICE_URL", "http://localhost:8090")
AUTH = os.environ.get("DASH_AUTH_SECRET", "change-me")

body = json.dumps({"order_id": "order-123", "amount_minor": 1000, "currency": "USD"}).encode()
req = urllib.request.Request(
    f"{SERVICE}/intents",
    data=body,
    headers={"Content-Type": "application/json", "X-Dash-Auth": AUTH},
    method="POST",
)

with urllib.request.urlopen(req) as resp:
    intent = json.load(resp)

print("Pay to:", intent["address"])
print("URI (QR):", intent["uri"])
# Persist intent["intent_id"] against your order; the webhook will reference it.

File: examples/node/create-intent.mjs
// Create a payment intent, then show intent.address / intent.uri to the customer.
//   node examples/node/create-intent.mjs
const SERVICE = process.env.DASH_SERVICE_URL ?? "http://localhost:8090";
const AUTH = process.env.DASH_AUTH_SECRET ?? "change-me";

const res = await fetch(`${SERVICE}/intents`, {
  method: "POST",
  headers: { "content-type": "application/json", "X-Dash-Auth": AUTH },
  body: JSON.stringify({ order_id: "order-123", amount_minor: 1000, currency: "USD" }),
});
if (!res.ok) throw new Error(`intent failed: HTTP ${res.status}`);

const intent = await res.json();
console.log("Pay to:", intent.address);
console.log("URI (QR):", intent.uri);
console.log("Expires:", intent.expires_at);
// Persist intent.intent_id against your order; the webhook will reference it.
